testing/fetcher.ts

```diff
@@ -230,6 +231,49 @@ export class AuthenticatingFetcher implements Fetcher {
           headers: {...headers, Authorization: `Basic ${encodedAuth}`},
         };
       }
+      case AuthenticationType.Custom: {
+        let urlWithSubstitutions = url;
+        let bodyWithSubstitutions = body;
+        let formWithSubstitutions = JSON.stringify(form);
+        let headersWithSubstitutions = JSON.stringify(headers);
+
+        const {params} = this._credentials as CustomCredentials;
+
+        Object.entries(params).forEach(([key, value]) => {
```

Comment:
I think we have enough test scaffolding in place that it shouldn't be awful to add a unittest case for this. Given the complexity here, worth having the coverage.

Reply:
done added unit tests